### PR TITLE
56168 Status Links Update

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR converts the VAOS navigation buttons to links per 508 review.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#56168

## Testing done
Unit testing

## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx [hunk 1]
```diff
@@ -46,55 +20,53 @@ export default function AppointmentListNavigation({ count, callback }) {
...
+              onClick={() => {
+                callback(true);
+                recordEvent({
+                  event: `${GA_PREFIX}-status-pending-link-clicked`,
+                });
+              }}
               // eslint-disable-next-line jsx-a11y/aria-proptypes
               aria-current={`${
-                location.pathname.endsWith('pending') ? 'page' : null
+                location.pathname.endsWith('pending') ? 'true' : false
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx:47]: Curious why is true set using single quote while false does not?

**Review by jenniemc — APPROVED** [review]
LGMT! Good idea replacing button with NavLink React component. Got one question...

File: src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx:47]: Good catch, once again!

**Comment by simiadebowale** [comment]
Waiting for RI to deploy. Everything looks good apart from Jennie's comments

**Comment by simiadebowale** [comment]
All links are read as "visited" by screen reader on first open.
Links are also purple which indicates they've been visited although they've not been visited by user.

https://user-images.githubusercontent.com/47654119/230508058-0c36f070-c807-4879-880c-5277b56436f8.mov

**Pull request merged**